sdk/servicebus/service-bus/src/receivers/shared.ts

```diff
@@ -226,3 +226,10 @@ function settleMessage(
     throw translateServiceBusError(err);
   });
 }
+
+export function numberOfEmptyIncomingSlots(receiver: any): number {
+  return (
+    (receiver as any).session._session.incoming.deliveries.capacity -
```

Comment:
Better types from "rhea" to avoid this machinery.

Reply:
To check - we're saying these better types are a pre-requisite? Or would we ship with this kind of check in place as it is?

Reply:
We can request rhea for it.
Waiting for rhea might be the best.

But just saying, rhea-promise does a bad job already..
![image](https://user-images.githubusercontent.com/10452642/110015682-00e9c580-7cd9-11eb-87fe-df5d904497f5.png)
We'd just be continuing the practice rhea-promise had started.

So, I guess it is fine to make this change independently.

Reply:
Any chance we could get `rhea-promise` to expose the APIs we need to do this calculation? I don't like that we're delving into the private properties of 2 libraries now.

Reply:
I can make the PR in `rhea-promise`, we can release the library and use them here I guess. That'd keep service-bus SDK cleaner.
@ramya-rao-a

Reply:
That sounds good to me

Reply:
https://github.com/amqp/rhea-promise/pull/79

Reply:
Please review the PR. Once new `rhea-promise` is released, I'll update this PR and take it out of draft.
Sounds good?

Reply:
Lets not wait for rhea-promise update to merge this PR

Reply:
https://github.com/amqp/rhea-promise/issues/85 Logged an issue in the rhea-promise since that's more appropriate and to resolve this thread.